### DISCUSSION TITLE
When files in the knowledge base change, clean up the files in S3

### DISCUSSION THREAD

**Discussion by majian159** [langgenius/dify#3863] [Suggestion]
Currently, when the knowledge base is deleted or documents within the knowledge base are deleted, the files already uploaded to S3 are not removed. This results in many redundant files, especially when identical files are repeatedly uploaded to S3.

**Comment by kerlion** [comment]
Yes, I found the same issue. The file is not deleted when it was deleted from the knowledge base.

**Comment by terrysun1216** [comment]
how to remove uploaded files in deleted knowledge?

**Comment by kyleli666** [comment]
I found the uploaded files are not used after indexed. So maybe A work around could be setting a s3 life cycle policy.
